src/main/kotlin/me/partlysanestudios/partlysaneskies/features/dungeons/TerminalWaypoints.kt

```diff
@@ -0,0 +1,171 @@
+package me.partlysanestudios.partlysaneskies.features.dungeons
+
+import me.partlysanestudios.partlysaneskies.PartlySaneSkies.Companion.minecraft
+import me.partlysanestudios.partlysaneskies.data.skyblockdata.IslandType
+import me.partlysanestudios.partlysaneskies.features.debug.DebugKey
+import me.partlysanestudios.partlysaneskies.render.waypoint.Waypoint
+import me.partlysanestudios.partlysaneskies.render.waypoint.WaypointEvent
+import me.partlysanestudios.partlysaneskies.render.waypoint.WaypointRenderPipeline
+import me.partlysanestudios.partlysaneskies.utils.ImageUtils.applyOpacity
+import me.partlysanestudios.partlysaneskies.utils.MathUtils.getDistance3d
+import me.partlysanestudios.partlysaneskies.utils.MinecraftUtils.getAllArmorStands
+import me.partlysanestudios.partlysaneskies.utils.SystemUtils.log
+import me.partlysanestudios.partlysaneskies.utils.vectors.Point3d
+import me.partlysanestudios.partlysaneskies.utils.vectors.Range3d
+import net.minecraft.util.IChatComponent
+import net.minecraftforge.client.event.ClientChatReceivedEvent
+import org.apache.logging.log4j.Level
+import java.awt.Color
+
+object TerminalWaypoints {
+
+    private val boundingBoxes = arrayOf(
+        Range3d(118.0, 103.0, 27.0, 90.0, 157.0,124.0),
+        Range3d(118.0, 103.0, 145.0, 19.0, 157.0, 118.0),
+        Range3d(-4.0, 103.0, 145.0, 19.0, 157.0, 50.0),
+        Range3d(-3.0, 103.0, 26.0, 92.0, 157.0, 53.0)
+    )
+    private val terminals = ArrayList<F7Puzzle>()
+
+    @WaypointEvent
+    fun onWaypointRender(pipeline: WaypointRenderPipeline) {
+        if (!IslandType.CATACOMBS.onIsland()) {
+            return
+        }
+        if (DebugKey.isDebugMode()) {
+            return
+        }
+        val playerPoint3d = Point3d.asPlayer()
+        val playerBoundingBox = getCurrentBoundingBox(playerPoint3d)
+
+
+
+        for (pointPair in getAllInactivePos()) {
```

Comment:
This and the active code is kinda duplicate,
is it possivle to make it mot duplcate?

Reply:
I really don't want to do this

Reply:
Okay then dont idk